python/GafferSceneUI/SceneInspector.py

```diff
@@ -1308,6 +1337,70 @@ def __labelButtonPress( self, label, event ) :
 		script = self.__target.scene.ancestor( Gaffer.ScriptNode )
 		GafferSceneUI.ContextAlgo.setSelectedPaths( script.context(), IECore.PathMatcher( [ label.getText() ] ) )
 
+class _ShaderParameterSection( LocationSection ) :
+
+	class __Inspector( Inspector ) :
+
+		def __init__( self, shaderAttribute, parameterName = None ) :
+
+			Inspector.__init__( self )
+
+			self.__shaderAttribute = shaderAttribute
+			self.__parameterName = parameterName
+
+		def name( self ) :
+
+			return self.__parameterName
+
+		def __call__( self, target ) :
+
+			parameters = self.__parameters( target )
+			if parameters is None :
+				return None
+
+			return parameters.get( self.__parameterName )
+
+		def children( self, target ) :
+
+			parameters = self.__parameters( target )
+			if parameters is None :
+				return []
+
+			return [ self.__class__( self.__shaderAttribute, parameterName = p ) for p in parameters.keys() ]
+
+		def __parameters( self, target ) :
+
+			if target.path is None :
+				return None
+
+			attributes = target.attributes()
+
+			shaders = attributes[self.__shaderAttribute]
```

Comment:
When doing a history traceback on a parameter value, I'm getting this error :

```File "/disk1/john/dev/build/gaffer/python/GafferSceneUI/SceneInspector.py", line 1378, in __parameters
    shaders = attributes[self.__shaderAttribute]
KeyError: 'ai:surface'
```

Looks like this is because the shader assignment doesn't exist upstream. If we were delegating to the attribute inspector to do the inspection, this would return `None` for this case and we'd be OK.

Reply:
Should work now. The rest of the changes helped.